src/Symfony/Component/Console/Terminal.php

```diff
@@ -64,14 +64,12 @@ public static function hasSttyAvailable(): bool
             return self::$stty;
         }
 
-        // skip check if exec function is disabled
-        if (!\function_exists('exec')) {
+        // skip check if shell_exec function is disabled
+        if (!\function_exists('shell_exec')) {
             return false;
         }
 
-        exec('stty 2>&1', $output, $exitcode);
-
-        return self::$stty = 0 === $exitcode;
+        return self::$stty = (bool) shell_exec('stty 2>&1');
```

Comment:
should be `stty 2> /dev/null` I think now

Reply:
This would change a bit the behavior, al least on windows (using GitBash):

Example code:
```
        exec('stty 2>&1', $output, $exitcode);
        var_dump(0 === $exitcode);
        var_dump((bool) shell_exec('stty 2> /dev/null'));
```

Result:
```
bool(true)
The System Cannot Find The Path Specified.
bool(false)
```

Reply:
what a mess!
Can you try this instead?
`command -v stty >/dev/null 2>&1 && stty 2> /dev/null`

Reply:
I found a way, working on windows. Please have a look if it's fine for you.